galsim/utilities.py

```diff
@@ -418,3 +418,133 @@ def _gammafn(x):
+                               directions or a list or tuple of two integers, containing the number
+                               of images in x and y directions respectively.
+    @param offsets             A list containing the offsets as galsim.PositionD instances
+                               corresponding to every image in `im_list'. The offsets must be spaced
+                               equally and must span an entire pixel area. The offset values must
+                               be symmetric around zero, hence taking positive and negative values,
+                               with upper and lower limits of +0.5 and -0.5 (limit values excluded).
+    @param add_flux            Should the routine add the fluxes of all the images (True) or average
+                               them (False)? [default:True]
+    @param suppress_warnings   Suppresses the warnings about the pixel scale of the output, if True.
+                               [default:False]
+    @param catch_offset_errors Checks for the consistency of `offsets` with `N` and raises Errors
+                               if inconsistencies found (True). Recommended, but could slow down
+                               the routine a little. [default:True]
+
+    @returns the interleaved image
+    """
+
+    if isinstance(N,int):
+        n1,n2 = N,N
+    elif hasattr(N,'__iter__'):
+        if len(N)==2:
+            n1,n2 = N
+        else:
+            raise TypeError("'N' has to be a list or a tuple of two integers")
+        if not (isinstance(n1,int) and  isinstance(n2,int)):
+            raise TypeError("'N' has to be of type int or a list/tuple of two integers")
+    else:
+        raise TypeError("'N' has to be of type int or a list/tuple of two integers")
+
+    if len(im_list)<2:
+        raise TypeError("'im_list' needs to have at least two instances of galsim.Image")
+
+    if (n1*n2 != len(im_list)):
+        raise ValueError("'N' is incompatible with the number of images in 'im_list'")
+
+    if len(im_list)!=len(offsets):
+        raise ValueError("'im_list' and 'offsets' must be lists of same length")
+    for offset in offsets:
+        if not isinstance(offset,galsim.PositionD):
+            raise TypeError("'offsets' must be a list of galsim.PositionD instances")
+
+    if isinstance(im_list[0],galsim.Image):
+        y_size, x_size = im_list[0].array.shape
+        scale = im_list[0].scale
+    else:
+        raise TypeError("'im_list' must be a list of galsim.Image instances")
+
+    for im in im_list[1:]:
+        if not isinstance(im,galsim.Image):
+            raise TypeError("'im_list' must be a list of galsim.Image instances")
+
+        if im.array.shape != (y_size,x_size):
+            raise ValueError("All galsim.Image instances in 'im_list' must be of the same size")
+ 
+        if im.scale != scale:
+            raise ValueError(
+                "All galsim.Image instances in 'im_list' must have the same pixel scale")
```

Comment:
Are you requiring them to have PixelScale WCS's?  Perhaps you want to take `wcs` from the first and check that `im.wcs != wcs` here.